validator_client/src/beacon_node_fallback.rs

```diff
@@ -427,6 +428,11 @@ impl<T: SlotClock, E: EthSpec> BeaconNodeFallback<T, E> {
                 match func(&$candidate.beacon_node).await {
                     Ok(val) => return Ok(val),
                     Err(e) => {
+                        debug!(
+                            log,
+                            "Request failed for candidate beacon node {}. BN candidate may not be ready.",
+                            $candidate.beacon_node.to_string(),
+                        );
```

Comment:
Lets make this error message a bit shorter, and put the extra info as params. Something like this (check that it compiles):

```suggestion
                        debug!(
                            log,
                            "Request to beacon node failed";
                            "node" => $candidate.beacon_node.to_string(),
                            "error" => ?e,
                        );
```

Reply:
No it doesn't, unfortunately. There's no type info for e in this context (macro def). 
```
`Err` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
```

Reply:
I can try and refactor the code to use a closure instead... @michaelsproul Do you see value in doing so?

Reply:
I don't think you need a closure, you could try adding an `Err: Debug` bound to the `where` clause

Reply:
Yes, that worked. 10x